java/pom.xml

```diff
@@ -100,19 +100,20 @@
             </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-assembly-plugin</artifactId>
+                <artifactId>maven-shade-plugin</artifactId>
```

Comment:
Without this change I was running into this [issue](https://github.com/grpc/grpc-java/issues/10826) when trying to run the examples locally. This change is based on the working sample from this [comment](https://github.com/grpc/grpc-java/issues/10826#issuecomment-1970047423). As far as I can tell everything is still working as expected for me with this change in place; with that said, it would be great to get others to pull this branch and test the examples to confirm these changes don't break anyone else